wasm/keystore.ts

```diff
@@ -0,0 +1,282 @@
+    private readonly core: WalletCore;
+    private readonly storage: IKeyStoreStorage;
+    private readonly walletIdsKey: string;
+
+    constructor(
+      core: WalletCore,
+      walletIdsKey: string,
+      storage: IKeyStoreStorage
+    ) {
+      this.core = core;
+      this.storage = storage;
+      this.walletIdsKey = walletIdsKey;
+    }
+
+    hasWallet(id: string): Promise<boolean> {
+      return this.storage
+        .get(id)
+        .then((wallet) => {
+          return wallet !== undefined;
+        })
+        .catch((error) => {
+          return false;
+        });
+    }
+
+    load(id: string): Promise<KeyStore.Wallet> {
+      return this.storage.get(id).then((wallet) => {
+        if (wallet === null) {
+          throw KeyStore.Error.WalletNotFound;
+        }
+        return wallet;
+      });
+    }
+
+    loadAll(): Promise<Wallet[]> {
+      return this.storage.loadAll(this.walletIdsKey);
+    }
+
+    delete(id: string, password: string): Promise<void> {
+      return this.storage.delete(id, password);
+    }
+
+    mapWallet(storedKey: StoredKey): KeyStore.Wallet {
+      let json = storedKey.exportJSON();
+      let wallet = JSON.parse(Buffer.from(json).toString()) as KeyStore.Wallet;
+      return wallet;
+    }
+
+    mapStoredKey(wallet: KeyStore.Wallet): StoredKey {
+      let json = Buffer.from(JSON.stringify(wallet));
+      let storedKey = this.core.StoredKey.importJSON(json);
+      return storedKey;
+    }
+
+    importWallet(wallet: KeyStore.Wallet): Promise<void> {
+      return this.storage.set(wallet.id, wallet);
+    }
+
+    import(
+      mnemonic: string,
+      name: string,
+      password: string,
+      coins: CoinType[]
+    ): Promise<KeyStore.Wallet> {
+      return new Promise((resolve, reject) => {
+        const { Mnemonic, StoredKey, HDWallet } = this.core;
+
+        if (!Mnemonic.isValid(mnemonic)) {
+          throw KeyStore.Error.InvalidMnemonic;
+        }
+        let pass = Buffer.from(password);
+        let storedKey = StoredKey.importHDWallet(
+          mnemonic,
+          name,
+          pass,
+          coins[0]
+        );
+        let hdWallet = HDWallet.createWithMnemonic(mnemonic, "");
+        coins.forEach((coin) => {
+          storedKey.accountForCoin(coin, hdWallet);
+        });
+        let wallet = this.mapWallet(storedKey);
+
+        storedKey.delete();
+        hdWallet.delete();
+
+        this.importWallet(wallet)
+          .then(() => {
+            resolve(wallet);
+          })
+          .catch((error) => {
+            reject(error);
+          });
+      });
+    }
+
+    importKey(
+      key: Uint8Array,
+      name: string,
+      password: string,
+      coin: CoinType
+    ): Promise<KeyStore.Wallet> {
+      return new Promise((resolve, reject) => {
+        const { StoredKey, PrivateKey, Curve } = this.core;
+
+        // FIXME: get curve from coin
+        if (
+          !PrivateKey.isValid(key, Curve.secp256k1) ||
```

Comment:
Is it all the curves we have? What about nist256p1?

Reply:
Good catch, this will be fixed in https://github.com/trustwallet/wallet-core/issues/2391